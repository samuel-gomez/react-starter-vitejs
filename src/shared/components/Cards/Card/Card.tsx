import type { ReactNode } from 'react';
import LasyImage from 'shared/components/LazyImage';
import { STORYBOOK, DESIGN_SYSTEM } from 'shared/constants';
import Icons from 'shared/components/Icons';

type TCardLink = {
  text: string;
  path: string;
};

export type TCard = {
  storybook?: TCardLink;
  designSystem?: TCardLink;
  title: ReactNode;
  picture: { name: string; alt: string };
};

const Card = ({ storybook, designSystem, title, picture: { name, alt } }: TCard) => (
  <article className="af-card af-skeleton">
    <LasyImage className="af-card__image" name={name} alt={alt} />
    <header className="af-card__header">
      <h3 className="af-card__title">{title}</h3>
    </header>
    <section className="af-card__content">
      <a
        aria-label="Voir le design system"
        className="btn af-btn--circle af-btn--svg"
        href={`${DESIGN_SYSTEM}${designSystem?.path ?? ''}`}
        title={storybook?.text ?? ''}
        target="_blank"
        rel="noopener noreferrer"
      >
        <Icons icon="slash" viewBox="0 0 342.988 280" />
      </a>
      <a
        aria-label="Voir le storybook"
        className="btn af-btn--circle af-btn--storybook"
        href={`${STORYBOOK}${storybook?.path ?? ''}`}
        title={storybook?.text ?? ''}
        target="_blank"
        rel="noopener noreferrer"
      >
        <Icons icon="storybook" viewBox="0 0 9.6 12" />
      </a>
    </section>
  </article>
);

export default Card;
