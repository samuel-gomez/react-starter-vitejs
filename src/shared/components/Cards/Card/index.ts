export { default } from './Card';
export type { TCard } from './Card';
