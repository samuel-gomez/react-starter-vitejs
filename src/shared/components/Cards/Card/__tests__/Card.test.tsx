import { screen, render, act, waitFor } from '@testing-library/react';
import Card from '../index';

const defaultProps = { title: 'title', picture: { name: 'accordion.svg', alt: 'alt_accordion' } };

describe('<Card/>', () => {
  it('Should render Card when image loaded', async () => {
    const { asFragment } = render(<Card {...defaultProps} />);
    await act(() => {
      screen.getByAltText('loading...');
    });

    await waitFor(() => expect(screen.getByAltText('alt_accordion')).toBeDefined());
    expect(asFragment()).toMatchSnapshot();
  });

  it('Should render Card when image loading', () => {
    const { asFragment } = render(<Card {...defaultProps} />);
    act(() => {
      screen.getByAltText('loading...');
    });
    expect(asFragment()).toMatchSnapshot();
  });
});
