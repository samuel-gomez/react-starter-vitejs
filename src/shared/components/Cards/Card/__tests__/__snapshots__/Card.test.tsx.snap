// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Card/> > Should render Card when image loaded 1`] = `
<DocumentFragment>
  <article
    class="af-card af-skeleton"
  >
    <img
      alt="alt_accordion"
      class="af-card__image"
      src="/public/images/accordion.svg"
    />
    <header
      class="af-card__header"
    >
      <h3
        class="af-card__title"
      >
        title
      </h3>
    </header>
    <section
      class="af-card__content"
    >
      <a
        aria-label="Voir le design system"
        class="btn af-btn--circle af-btn--svg"
        href="https://axafrance.github.io/design-system/"
        rel="noopener noreferrer"
        target="_blank"
        title=""
      >
        <svg
          aria-labelledby="title"
          class="af-icon-svg"
          role="img"
          viewBox="0 0 342.988 280"
        >
          <title>
            slash
          </title>
          <path
            d="M233.869 0h109.119L109.119 280H0L233.869 0zM299.988 144.545V239h-78.797l-34.203 41h156V93l-43 51.545zM43 41h78.797L156 0H0v187l43-51.545V41z"
          />
        </svg>
      </a>
      <a
        aria-label="Voir le storybook"
        class="btn af-btn--circle af-btn--storybook"
        href="https://axafrance.github.io/react-toolkit/latest/storybook/?path=/story/"
        rel="noopener noreferrer"
        target="_blank"
        title=""
      >
        <svg
          aria-labelledby="title"
          class="af-icon-svg"
          role="img"
          viewBox="0 0 9.6 12"
        >
          <title>
            storybook
          </title>
          <path
            d="M9.6.6C9.6.2 9.3 0 9 0h-.7v1.4s0 .1-.1.1h-.1l-.3-.4-.5.4h-.1v-.1L7.3 0 .6.5c-.3 0-.6.3-.5.6L.4 11c0 .3.3.6.6.6l8 .4c.3 0 .6-.3.6-.6V.6zM5.1 9.7c-1.4 0-2.7-.6-2.5-2.6.1-.2 1.9-.2 1.9 0 0 .8.2 1.1.6 1.1.4 0 .5-.2.5-.5 0-1.2-3-1.3-3-3.5C2.6 2.9 3.5 2 5 2c1.6 0 2.4.8 2.4 2.4-.2.2-1.8.3-1.8 0 0-.9-.4-.9-.6-.9-.2 0-.6.1-.6.5 0 1.2 3 1.1 3 3.5.1 1.4-.9 2.2-2.3 2.2z"
          />
        </svg>
      </a>
    </section>
  </article>
</DocumentFragment>
`;

exports[`<Card/> > Should render Card when image loading 1`] = `
<DocumentFragment>
  <article
    class="af-card af-skeleton"
  >
    <img
      alt="loading..."
      class="af-lasyimage"
      src="/src/assets/puff.svg"
    />
    <header
      class="af-card__header"
    >
      <h3
        class="af-card__title"
      >
        title
      </h3>
    </header>
    <section
      class="af-card__content"
    >
      <a
        aria-label="Voir le design system"
        class="btn af-btn--circle af-btn--svg"
        href="https://axafrance.github.io/design-system/"
        rel="noopener noreferrer"
        target="_blank"
        title=""
      >
        <svg
          aria-labelledby="title"
          class="af-icon-svg"
          role="img"
          viewBox="0 0 342.988 280"
        >
          <title>
            slash
          </title>
          <path
            d="M233.869 0h109.119L109.119 280H0L233.869 0zM299.988 144.545V239h-78.797l-34.203 41h156V93l-43 51.545zM43 41h78.797L156 0H0v187l43-51.545V41z"
          />
        </svg>
      </a>
      <a
        aria-label="Voir le storybook"
        class="btn af-btn--circle af-btn--storybook"
        href="https://axafrance.github.io/react-toolkit/latest/storybook/?path=/story/"
        rel="noopener noreferrer"
        target="_blank"
        title=""
      >
        <svg
          aria-labelledby="title"
          class="af-icon-svg"
          role="img"
          viewBox="0 0 9.6 12"
        >
          <title>
            storybook
          </title>
          <path
            d="M9.6.6C9.6.2 9.3 0 9 0h-.7v1.4s0 .1-.1.1h-.1l-.3-.4-.5.4h-.1v-.1L7.3 0 .6.5c-.3 0-.6.3-.5.6L.4 11c0 .3.3.6.6.6l8 .4c.3 0 .6-.3.6-.6V.6zM5.1 9.7c-1.4 0-2.7-.6-2.5-2.6.1-.2 1.9-.2 1.9 0 0 .8.2 1.1.6 1.1.4 0 .5-.2.5-.5 0-1.2-3-1.3-3-3.5C2.6 2.9 3.5 2 5 2c1.6 0 2.4.8 2.4 2.4-.2.2-1.8.3-1.8 0 0-.9-.4-.9-.6-.9-.2 0-.6.1-.6.5 0 1.2 3 1.1 3 3.5.1 1.4-.9 2.2-2.3 2.2z"
          />
        </svg>
      </a>
    </section>
  </article>
</DocumentFragment>
`;
